korman/exporter/animation.py

```diff
@@ -351,20 +358,62 @@ def _convert_transform_animation(self, bo, fcurves, xform, *, allow_empty: Optio
         channel = plMatrixControllerChannel()
         channel.controller = tm
         applicator.channel = channel
-        channel.affine = utils.affine_parts(xform)
+        channel.affine = utils.affine_parts(default_xform)
 
         return applicator
 
-    def convert_transform_controller(self, fcurves, rotation_mode: str, xform, *,
+    def convert_transform_controller(self, fcurves, rotation_mode: str, default_xform, adjust_xform, *,
                                      allow_empty: Optional[bool] = False,
                                      start: Optional[int] = None,
                                      end: Optional[int] = None) -> Union[None, plCompoundController]:
         if not fcurves and not allow_empty:
             return None
 
-        pos = self.make_pos_controller(fcurves, "location", xform.to_translation(), start=start, end=end)
-        rot = self.make_rot_controller(fcurves, rotation_mode, xform, start=start, end=end)
-        scale = self.make_scale_controller(fcurves, "scale", xform.to_scale(), start=start, end=end)
+        if adjust_xform is not None:
+            # We have to edit the keyframes to make the anim local..
+            # In many cases this should work fine, but sometimes scale and rotation might
+            # still cause issues. Also, euler angles need to be converted to quaternion
+            # and back to eulers, which could cause issues. Not much we can do about it.
+            adjust_rotation = adjust_xform.to_quaternion()
+            adjust_scale = adjust_xform.to_scale()
+
+            # Helpers to adjust keyframes in case animation is not local (adjust_xform == identity)
+            def convert_pos_keyframe(pos):
+                # Position: can transform to local space without issues.
+                return tuple(adjust_xform * mathutils.Vector(pos))
+
+            def convert_rot_keyframe(rot):
+                # Rotation: may cause issues if scale is present.
+                if isinstance(rot, mathutils.Quaternion): # quaternion from an axis-angle
+                    return adjust_rotation * rot
+                elif isinstance(rot, mathutils.Euler):
+                    return (adjust_rotation * rot.to_quaternion()).to_euler(rot.order)
+                else: # tuple
+                    if len(rot) == 4: # quat in a tuple
+                        return (adjust_rotation * mathutils.Quaternion(rot))[:]
+                    else: # XYZ euler in a tuple
+                        rot = mathutils.Euler(rot, "XYZ").to_quaternion()
+                        return (adjust_rotation * rot).to_euler("XYZ")[:]
+
+            def convert_scale_keyframe(scale):
+                # Scale: very likely to cause issues.
+                return (a * b for a, b in zip(adjust_scale, scale))
```

Comment:
```suggestion
                return [a * b for a, b in zip(adjust_scale, scale)]
```

I think it would be best to use a list comprehension here. The underlying keyframe processing code expects sequences, not generators 😉